.github/workflows/pre-build-validation.yml

```diff
@@ -27,7 +33,38 @@ jobs:
         repository: ${{ github.event.inputs.owner }}/${{ github.event.inputs.repo }}
         ref: ${{ github.event.inputs.branch }}
         path: repo
+        fetch-depth: 0
+    - name: Check for Product Changes
+      id: pull_request
+      run: |
+        cd repo
+        yamlfiles=$(git diff --name-only origin/main -- mmv1/products)
```

Comment:
This is tricky.. We don't want to diff with origin main here because it will include changes that aren't in the base PR. This sha is forked from an earlier version of origin/main.

Ideally we want to set up the code base branches that we diff between the manually.

Check out how generate diff sets up the repository branches to prepare for build. I image we want to do something similar.

https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/gcb-generate-diffs-new.yml#L40-L65

Reply:
Could you expand on this? I agree merging the main branch will be necessary. Essentially thats what the code in generate diffs does. I don't think relying on the developer to have latest main to be pulled is reliable

Reply:
I added a line to the script that merges origin/main into the current branch.

Reply:
Do you know what the UX is in case of merge failure? Could you test this?

Reply:
I'm not sure how to set up a test for this, but the result should be yaml linting failing to run when there are merge conflicts with main.

Reply:
This is assuming a failed merge doesn't prompt an interactive shell environment. You could base a pr off an old commit and then try change a file that the next, merged commit changes. You should hit this scenario then. Would rather know what the UX is then have this happen in the wild.

Reply:
After testing it looks like the diff works as expected when there's a merge conflict. The merge puts conflict markers in the relevant files, which show up as diffs to `git diff`.

Reply:
mm okay gotcha! so if I understand correctly then there is no issue with the linter as long as the conflicts aren't in the linted files